client/src/explorer_common/explorer_components.js

```diff
@@ -181,7 +181,9 @@ export const ExplorerNode = ({
                 }
               >
                 {_.isFunction(val_display)
-                  ? val_display(get_val(node))
+                  ? depth === 0
+                    ? val_display(get_val(node), node)
+                    : val_display(get_val(node))
```

Comment:
No reason `depth === 0` should be special here, always pass `node` as the second arg. Up to `val_display` to check depth and own any related logic.